include/quda_internal.h

```diff
@@ -1,6 +1,13 @@
 #pragma once
 
+#include <quda_define.h>
 #include <quda_api.h>
+
+#if defined(QUDA_TARGET_CUDA)
```

Comment:
Can this move to target?

Reply:
This actually cannot be moved to target, or at least not cleanly yet.  This is partly by design.  The target include directly, e.g. `include/target/cuda`, is not visible to external projects, in order to ensure that apps that link to QUDA are architecture independent.  However, the tests do require, at least at present, to be able to see the target system includes, `cuda.h`.

This is something I will be revisiting after this PR, and indeed Balint has one potential solution in his HIP PR, that we could adopt.  I wanted to mull on this first though.